src/interfaces.ts

```diff
@@ -24,48 +21,23 @@ export interface ConnectionOptions {
 }
 
 export interface RpcParams {
-  controller?: boolean;
-  createStatement?: string;
-  description?: string;
-  dryrun?: boolean;
-  statement?: string;
-  tableId?: string;
-  txnHash?: string;
-}
-
-export interface RpcRequestParam {
   controller?: string;
   /* eslint-disable-next-line camelcase */
   create_statement?: string;
-  description?: string;
-  dryrun?: boolean;
-  id?: string;
   statement?: string;
   /* eslint-disable-next-line camelcase */
   txn_hash?: string;
 }
 
-export interface ConnectionReceipt {
-  jwsToken: Token;
-  ethAccounts: Array<string>;
-}
-
 export interface SupportedNetwork {
   key: string;
   phrase: string;
   chainId: number;
 }
 
-/**
- * ColumnDescriptor gives metadata about a colum (name, type)
- */
-export interface ColumnDescriptor {
-  name: string;
-}
-
 export type KeyVal<T = any> = [string, T];
 
-export type Column = Array<ColumnDescriptor>;
+export type Column = Array<{ name: string }>;
 
 export type Row = Array<string | number | boolean>;
 
```

Comment:
`ReadQueryInterface` could be made generic in rows. Such that:

```
export interface ReadQueryResult<Row extends Array<string | number | boolean> = Array<any>> {
  columns: Array<Column>;
   rows: Row;
 }
```

And we delete the `Row` type above (because it is no longer needed).

Does that seem useful?

Reply:
Just a suggestion to make read generic, so you can do things like:

```
type MyType = [number, string, number]
const response = await sdk.read<MyType>("blah blah")
```

where response contains an array of rows that are typed as MyType.

Reply:
Love this!